doxygen/contributor_help_pages/adding_new_distributions.md

```diff
@@ -0,0 +1,460 @@
+
+Note that for univariate distributions wolfram handles things a bit simpler,
+though for multivariate distributions you'll have to use matrixcalculus. Though
+for both you'll have a much nicer time if you plug in the log'd version of the
+function. One other nice thing about the matrixcalculus site is that it can generate latex
+which is nice for documentation.
+
+
+$$
+\begin{aligned}
+f = \text{Normal}(y|\mu,\sigma) &= -\frac{1}{2} \left(\frac{y-\mu}{\sigma}\right)^2 - \log(\sigma) - \frac{1}{2}\log(2\pi) \cr
+\frac{\partial f}{\partial y} &= -\frac{y-\mu}{\sigma^{2}} \cr
+\frac{\partial f}{\partial \mu} &= \frac{y-\mu}{\sigma^{2}} \cr
+\frac{\partial f}{\partial \sigma} &= -\frac{1}{\sigma} - \frac{(y-\mu)^{2}}{\sigma^{3}}
+\end{aligned}
+$$
+
+It's a little early, but once we get the `lpdf` function working with the above we will want to get out a pen and paper to simplify and find common subexpressions we only need to calculate once.
+For instance in the normal we can compute `y - mu` and `1/sigma`
+
+$$
+\begin{aligned}
+f(y|\mu,\sigma) = \text{Normal}(y|\mu,\sigma) &= -\frac{1}{2} \left(\frac{y-\mu}{\sigma}\right)^2 - \log(\sigma) - \frac{1}{2}\log(2\pi) \cr
+\frac{\partial f}{\partial y} &= -t_3 \cr
+\frac{\partial f}{\partial \mu} &= t_3 \cr
+\frac{\partial f}{\partial \sigma} &= t_3 \cdot t_0 - t_0 \cr
+\text{Where} \cr
+t_0 &= \frac{1}{\sigma} \cr
+t_1 &= t_{0}^2 \cr
+t_2 &= y - \mu \cr
+t_3 &= \frac{t_2}{t_1}
+\end{aligned}
+$$
+
+
+### Writing the function
+
+
+So now let's add the lpdf function in `stan/math/prim/dist/new_normal_lpdf.hpp`.
+First we'll go over what we have to do before we start doing any math. We'll be
+breaking down Stan's current `normal_lpdf` function which you can find [here](https://github.com/stan-dev/math/blob/develop/stan/math/prim/prob/normal_lpdf.hpp).
+
+
+#### Distribution Signature
+
+```cpp
+/** \ingroup prob_dists
+ * Docs describing the templates and arguments
+ */
+template <bool propto, typename T_y, typename T_loc, typename T_scale>
+inline return_type_t<T_y, T_loc, T_scale> normal_lpdf(const T_y& y,
+                                                      const T_loc& mu,
+                                                      const T_scale& sigma) {}
+```
+
+Each of the input arguments represent the inputs to the `Normal` function we wrote out above.
+The template parameters for univariate distributions are very general and they
+must work for all of Stan's scalar types
+`double`, `var`, and `fvar<T>` while accepting mixtures of scalars and vectors.
+The return of the function is the joint log probability accumulated over all
+of the inputs which is a scalar of the least upper bound of all the
+parameters scalar types. That's a lot of big words, but in essense means that
+if one of the inputs is a `var` and the others are double the return type needs to be
+a `double`. If the input signature contained `fvar<var>`, `var`, `double` then the
```

Comment:
Should this be "if one of the inputs is a `var` and the others are `double` the return type needs to be a `var`"?

Reply:
Lol yes extremely whoops on my part